docs/source/driver/snowflake.rst

```diff
@@ -469,6 +469,10 @@ These options map 1:1 with the Snowflake `Config object <https://pkg.go.dev/gith
     non-zero scaled columns will be returned as ``Float64`` typed Arrow columns.
     The default is ``true``.
 
+``adbc.snowflake.sql.client_option.use_max_microseconds_precision``
+    When ``true``, nanoseconds will be converted to microseconds
+    to avoid the overflow of the Timestamp type. Does not affect the Time32 or Time64 types.
```

Comment:
(1) Can we explicitly list what types it _does_ apply to?
(2) Is there any chance we will want to pick seconds/milliseconds instead of microseconds ever?

Reply:
It’s just an arrow.Timestamp issue. Are you thinking of calling out the different Snowflake types?

I started down another path of trying to specify the precision type in combination with a context change (outlined in https://github.com/snowflakedb/gosnowflake/issues/1430) but that didn’t work.

I settled on microsecond precision as a similar pattern to the useHighPrecision setting so the caller can specify the desired behavior.

Reply:
And yes, but my question is whether a boolean is sufficient here